posts/2020-01-30-Rust-1.41.0.md

```diff
@@ -0,0 +1,211 @@
+`Box<T>` where the C definition is just using `T*` can lead to undefined
+behavior.
+
+To read more, [consult the documentation for `Box<T>`][box_docs].
+
+### Library changes
+
+[`Result::map_or`]: https://doc.rust-lang.org/std/result/enum.Result.html#method.map_or
+[`Result::map_or_else`]: https://doc.rust-lang.org/std/result/enum.Result.html#method.map_or_else
+[`Option::map_or`]: https://doc.rust-lang.org/std/option/enum.Option.html#method.map_or
+[`Option::map_or_else`]: https://doc.rust-lang.org/std/option/enum.Option.html#method.map_or_else
+[`std::rc::Weak::weak_count`]: https://doc.rust-lang.org/std/rc/struct.Weak.html#method.weak_count
+[`std::rc::Weak::strong_count`]: https://doc.rust-lang.org/std/rc/struct.Weak.html#method.strong_count
+[`std::sync::Weak::weak_count`]: https://doc.rust-lang.org/std/sync/struct.Weak.html#method.weak_count
+[`std::sync::Weak::strong_count`]: https://doc.rust-lang.org/std/sync/struct.Weak.html#method.strong_count
+[pr_66277]: https://github.com/rust-lang/rust/pull/66277
+[pr_65013]: https://github.com/rust-lang/rust/pull/65013
+
+In Rust 1.41.0, we've made the following additions to the standard library:
+
+- The [`Result::map_or`] and [`Result::map_or_else`] methods were stabilized.
+
+  Similar to [`Option::map_or`] and [`Option::map_or_else`], these methods are
+  shorthands for the `.map(|val| process(val)).unwrap_or(default)` pattern.
+
+- [`NonZero*` numerics now implement `From<NonZero*>` if it's a smaller integer
+  width.][pr_66277] For example, `NonZeroU16` now implements `From<NonZeroU8>`.
+
+- The `weak_count` and `strong_count` methods were stabilized.
+
+    - [`std::rc::Weak::weak_count`]
+    - [`std::rc::Weak::strong_count`]
+    - [`std::sync::Weak::weak_count`]
+    - [`std::sync::Weak::strong_count`]
+
+  These methods return the number of weak (`rc::Weak<T>` and `sync::Weak<T>`)
+  or strong (`Rc<T>` and `Arc<T>`) pointers to the allocation respectively.
+
+- [`MaybeUninit<T>` now implements `fmt::Debug`.][pr_65013]
+
+### Reducing support for 32-bit Apple targets soon
+
+Rust 1.41.0 is the last release with the current level of compiler support for
+32-bit Apple targets, including the `i686-apple-darwin` target. Starting from
+Rust 1.42.0, these targets will be demoted to the lowest support tier.
+
+[You can learn more about this change in this blog post.][32bit-demotion]
+
+[32bit-demotion]: https://blog.rust-lang.org/2020/01/03/reducing-support-for-32-bit-apple-targets.html
+
+### Other changes
+
+[relnotes-cargo]: https://github.com/rust-lang/cargo/blob/master/CHANGELOG.md#cargo-141-2020-01-30
+[relnotes-clippy]: https://github.com/rust-lang/rust-clippy/blob/master/CHANGELOG.md#rust-141
+
+There are other changes in the Rust 1.41.0 release: check out what changed in
+[Rust][notes], [Cargo][relnotes-cargo], and [Clippy][relnotes-clippy]. We also
+have started landing MIR optimizations, which should improve compile times. For
+more, read [mir-opt].
```

Comment:
I don't think we ever put this link in. Should be https://blog.rust-lang.org/inside-rust/2019/12/02/const-prop-on-by-default.html - on a phone so can't push right now.

Reply:
Fixed.